packages/story-editor/src/components/library/panes/media/local/hotlink/hotlinkModal.js

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { __, sprintf, translateToExclusiveList } from '@web-stories-wp/i18n';
+import { Input } from '@web-stories-wp/design-system';
+import { useState, useCallback, useRef, useLayoutEffect } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { getFileExtFromUrl } from '@web-stories-wp/media';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../../../../app';
+import Dialog from '../../../../../dialog';
+import { withProtocol } from '../../../../../../utils/url';
+import useInsert from './useInsert';
+
+const InputWrapper = styled.div`
+  margin: 16px 4px;
+  width: 470px;
+  height: 100px;
+`;
+
+function HotlinkModal({ isOpen, onClose }) {
+  const { allowedFileTypes } = useConfig();
+  const [errorMsg, setErrorMsg] = useState(false);
+  const inputRef = useRef(null);
+
+  useLayoutEffect(() => {
+    // Wait one tick to ensure the input has been loaded.
+    const timeout = setTimeout(() => {
+      if (isOpen && inputRef.current) {
+        inputRef.current.focus();
+      }
+    });
+    return () => clearTimeout(timeout);
+  }, [isOpen, inputRef]);
+
+  // @todo We're not really uploading anything here, so should we have a fixed list instead?
+  let description = __('No file types are currently supported.', 'web-stories');
+  if (allowedFileTypes.length) {
+    description = sprintf(
+      /* translators: %s is a list of allowed file extensions. */
+      __('You can insert %s.', 'web-stories'),
+      translateToExclusiveList(allowedFileTypes)
+    );
+  }
+  const error = sprintf(
+    /* translators: %s is the description with allowed file extensions. */
+    __('Invalid link. %s', 'web-stories'),
+    description
+  );
+  const [link, setLink] = useState('');
+
+  const getFileInfo = useCallback(
+    (value = link) => {
+      const ext = getFileExtFromUrl(value);
+      let type = null;
+      if (!allowedFileTypes.includes(ext)) {
+        setErrorMsg(error);
+      } else {
+        setErrorMsg(null);
+        type = ['m4v', 'mp4', 'webm'].includes(ext) ? 'video' : 'image';
+      }
```

Comment:
Let's add a todo to get this type from server-side validation instead. Then we can get rid of `getFileExtFromUrl`

Reply:
Hm, @swissspidy what do you mean by "instead"? We still need to get the type until we don't have the server-side validation, no?. Or perhaps I'm misunderstanding what you mean.

Reply:
Once we have some server-side logic, `onInsert` can simply:

1. Pass the URL to the server
2. Retrieve information about the file from the server (type, URL, size, etc.)
3. If type is invalid, inform user
4. Else, generate poster, insert media, etc.

No need for `getFileExtFromUrl` (which doesn't work for extension-less URLs anyway) and client-side mime type checks